### ISSUE TITLE
Loop Field Type

### ISSUE THREAD

**Issue by viruthagiri** [pods-framework/pods#109]
<em>Description modified September 1st, 2015 and July 4th, 2022 by @sc0ttkclark</em>

### Feature Summary

A loop field will be a field that contains other fields. For example a Pod called 'book' could have a loop field called 'chapters', which had its own set of fields such as, 'chapter_title', 'chapter_content', 'chapter_start_page' and 'chapter_end_page'. An item in the book Pod, could have as many entires in the chapters field as necessary, and each one would have its own set of fields, with their own values.

A lot of planning has gone into this feature, but its implementation has been delayed. Performance improvements that we plan to address in 2.3.19 and 3.0 have been prioritized over loop fields as they are essential for proper implementation of loop fields.

### Work to be done (needs to be revisited)
- [x] Design Loop Field workflows (@curtismchale)
- [x] Mockup Loop Field workflows (@curtismchale)
- [x] Develop prototype, minus JS for fields (@curtismchale)
- [ ] Refactor and clean up CMB2 structures to support nested fields (@pglewis, @sc0ttkclark)
- [ ] Refactor and clean up field type JS for group fields to support better nesting (@jtsternberg, @sc0ttkclark)
- [ ] Test and finish up saving process and storage handling for table/meta storage types (@sc0ttkclark)
- [ ] Test and finish up field() / display() for getting data out of loop field (@sc0ttkclark)
- [ ] Add field traversal for loop field data arrays (@sc0ttkclark)
- [ ] Create Unit tests (@pglewis, @sc0ttkclark)
- [ ] Write Documentation page (@shelob9)

We are actively seeking contributors for any of the open tasks above, please hop in and help out if you're interested! If you are unsure of how to get started or have any questions, feel free to ask @sc0ttkclark here or at http://pods.io/chat/) for help getting started.
### Original issue summary

Hello pods team,
I'm using magic fields 2.0 which contains features like "add another field" and "add another group" type of fields. 
Yesterday one of my internet friend recommended pods 2.0. I tested it and i really love it. Magic fields 2.0 stores the data as metadata. I'll get performance issue. Thats why i would like to use pods 2.0. Pods satisfies me everything except that "add another field" and "add another group" type of field. 

Here is the snapshot of what i'm asking 
Before duplicating the field:
http://wiki.magicfields.org/lib/exe/fetch.php?media=v2_field_duplicate_post.jpg

After duplicating the field:
http://wiki.magicfields.org/lib/exe/fetch.php?media=v2_field_duplicate_complete.jpg

Before duplicating the group:
http://wiki.magicfields.org/lib/exe/fetch.php?media=v2_group_duplicate_post.jpg

After duplicating the group:
http://wiki.magicfields.org/lib/exe/fetch.php?media=v2_group_duplicate_remove.jpg   

Does pods 2.x has that feature and as a pods newbie am i missing something?

**Comment by StaggerLeee** [comment]
Whai is the deal with Advanced Custom Fields plugin ?
It is literally one man (show) plugin. Is he some kind of coding genius, or what.
He makes it look so easy.

Hope not from colleagues developers to publicly valuate quality of ACF code. But whole plugin is still enigma for me.
Seems as Pods struggle so much and he does it very "easy".

Dont want to offend someone. I appreciate all work you done in Pods. Just wondering because I made long time ago fatal mistake and all my websites was dependent of K2 exstension for Joomla. 

Now it is the same with ACF. If plugin dissapear troubles again, not easy to convert data.

You can ask why writing about this here. Reason is ACF has repeater field (loop) since ages ago and working very well. Groups too, and much more. Why you struggle so much to implement it in Pods ?

**Comment by pixelkicks** [comment]
Last comment from Scott was in December, saying it's not far off.  Are we any closer?

**Comment by sc0ttkclark** [comment]
Sure, I can definitely provide an update here --

We've continued to have our hands full with everything ongoing in Pods 2.5.x. Right now we've got multiple things going between @Shelob9 @pglewis @jimtrue @NikV and I.

@pglewis is working on Term splitting for WP 4.2 compatibility, we're going the extra mile versus some other plugins that I've seen. After he's done with that, he'll be switching back to merging Pods 2.5.x into Pods 3.0 via #2822 which has been a big holdup on any 3.0 progress.

@Shelob9 is working on our REST API and trying to get it ready for the upcoming version change from the WP REST API folks.

@NikV is continuing to work on docs and moving forward GitHub fixes and enhancements on 2.5.x and 3.0. We setup a new requirement designed to prevent focus getting lost on 3.0 which requires all Pull Requests for Pods 2.5.x to have a Pods 3.0 patch as well before we merge. He's been doing that for doc issues and his other tasks.

Both @Shelob9 and @jimtrue are still maintaining our support queues across pods.io, wordpress.org, and our free Slack support chat. They both have also been running PodsCasts since February (http://pods.io/category/podscast/) and trying to help reduce their support load so we can focus on the code more.

We just recently launched Friends of Pods (https://pods.io/friends-of-pods/) last month, almost a month to this day. So far it's helped us get funding to let us rely less on client-projects. Up to then, @pglewis has been in a continuous series of client projects that have kept him away from helping out on Pods core much. He almost solely has been responsible for keeping the lights on, where as our Automattic sponsorship covers the other half of the bills. Friends of Pods has so far brought in more recurring donations than the we've seen in one-time donations in more than 7 months alone. I'd consider it a success, but we still have further to go -- we're not completely free of client projects and Phil is still only able to focus on Pods 3.0 tasks until we get 3.0 out the door.

Me? I've been working on helping @pglewis with the merge, various support help with @jimtrue and @Shelob9, helping out on PodsCasts with example code, getting unit tests running on Pods 2.5.x and Pods 3.0. Helping @pglewis on Term splitting. I've on hold working on the CMB2 merge which has it's own loop fields until we can finish the 2.x merge into 3.0 so we don't lose out on all the fixes and enhancements. CMB2 merge will take out a bunch of things we no longer need, but that would put a massive block on merging things that we still need so that's been prioritized under the 3.0 merge itself. Loop fields CSS/JS needs merging in, but we're also still trying to figure out how best to take advantage of everything that CMB2 gives us in form UI for that. That's along with other GitHub tweaks I've been a part of, reviewing code coming in, keeping the lights on, and planning for WordCamp DFW 2015 and a possible PodsCamp in 2015 (still not decided yet).

I can still tell you that Loop Fields will be in Pods 3.0, it's just held up by a number of tasks ongoing for Pods 2.5.x (for WP 4.2) and Pods 3.0 that would be much more time consuming post-loop-fields work.

If you want to help with code, hit us up in our #dev-core Slack channel. If you want to help give us the resources to keep hyper focused on Pods 3.0 like we've been this past month, consider becoming a Friend of Pods -- https://pods.io/friends-of-pods/ -- either as a one-time donation or recurring member.

More soon, most of our hold-ups are now on the cusp of completion: 2.x >> 3.0 merge, CMB2 merge, then Loop Fields. Our team continues to be devoted to keeping Pods going and we're all doing our part. It may not look like it at times, but you'd be amazed at how many things can keep something like Loop Fields delayed, either intentionally or unintentionally -- like when developers need bug fixes in Pods 2.x or something changes in a WP release like in 4.2 that requires a significant time investment.

**Comment by sc0ttkclark** [comment]
I also did want to give @jamesgol major props, his efforts the past many months have been a godsend. I'd go into more details about everything he's been doing, but I don't think I have enough time to gather a list of all of the notable things he's contributed to Pods.

**Comment by StaggerLeee** [comment]
Thanks for clarification and all good work.

Dont mind if I mention another plugin her. I know comparations like this are not popular in developer communities.

Despite I respect ACF developer a lot, bought a PRO licence, i would like to get rid of this dependency on my websites. No offence to developer but he could have in his life some unexpected events that prevent him developing plugin anymore.

There are many of you guys behind Pods, and would like to switch to Pods at whole. Use nothing more except it.

**Comment by lkraav** [comment]
@StaggerLeee do you mind providing a URL reference to info on what trouble ACF's Elliott is going through? Not looking to go off topic here, but I'm also trying to reduce my bus factor by supporting Pods.

**Comment by StaggerLeee** [comment]
Not any now, as I know it. But you know how it goes with "one man plugin(s)".
This is not gallery popup plugin. When it reach its end just delete it and replace it with another.
Or to take more advanced example. This is not even Jetpack, delete Jetpack and replace with other plugins.

Websites are much dependent on AFC and dont want some day go through all websites and fix them for hours and days. If it would be easy fix at all.

I had one bad experience. if it was not for one Frenchman and his 2 commercial plugins I would never be able to convert my Joomla (K2) websites to WordPress. I had more luck than knowledge there.

**Comment by sc0ttkclark** [comment]
It also doesn't hurt that many of our main team are active contributors on WordPress core.

Also there's this project I'm working on moving forward as a feature proposal (developer focused):

https://github.com/sc0ttkclark/wordpress-fields-api

**Comment by szepeviktor** [comment]
Is the nature of the bounty public?

**Comment by quasel** [comment]
lable removed ^^ but take a look at https://www.bountysource.com/teams/pods-framework

**Comment by szepeviktor** [comment]
I know that it takes hard work and long night to check those boxes at the top.
Do you have any plans for the progress?

**Comment by sc0ttkclark** [comment]
This is in progress as part of the CMB2 integration, but first we're getting Pods 3.0 passing all of our unit tests so we have a good basis for what we break during the CMB2 integration part.

**Comment by szepeviktor** [comment]
Thank you.

**Comment by kelfy** [comment]
Great. So looking forward to it

**Comment by sc0ttkclark** [comment]
Updated the main to-do task list in the main description for CMB2 tasks

**Comment by visualdatasolutions** [comment]
Hi there. 

I needed the grouping features in some of my projects, and recently got the idea of defining groups by the sequence of custom fields ( menu_order of fields in wp_posts table), using relationship fields named after a pattern to separate groups. This is similar to reading a table of contents from a printed book, when we see a line starting with the word "Chapter" we KNOW it is a chapter and treat it as such. We can instruct the computer to do the same.

I have uploaded the code and some screen shots to WordPress repository as proof of concept, see this link below. This is not loop field and may not suit everyone's needs. I am writing this for brainstorming as a interim work-around, and for seeing whether it fits in the big picture of future development of Pods.

[Grouping of custom fields in PODS](https://wordpress.org/plugins/custom-field-groups-for-pods/screenshots/)

![screenshot-3](https://cloud.githubusercontent.com/assets/13419927/9754085/a25cd698-5708-11e5-926b-a1c4f1b14533.png)

![front end display of group heading](https://ps.w.org/custom-field-groups-for-pods/assets/screenshot-5.png?rev=1238771)

Any field type can be used to define groups. I have chosen relationship type as the conditions to display the group  can be stored in the relationship field setting. So in a WP loop, we can retrieve details of the current post and decide whether to show the group of fields. similar to location rules found in ACF. 

The grouping is done using existing pods field structure and storage structure.  The query for groups returns a multi-dimensional array of groups and fields contained within, along with the field settings. I 
have used another field (footer) named after another pattern to define subgroups within a group.

the groups so defined can be used in pods short code and magic tags by adding to filters found in the functions in charge of parsing short code and magic tags. this is already done as shown in the screen shots.

When people produces a form containing all fields, the special fields (headers and footers) will produce a label only without input cells. 

$pod->display($header) will show nothing as header fields are definition only without storing any value. It should be easy to implement by adding another function in pods class to display all the fields within a group. 

the query result for groups can be cached for better performance. this is on my to do list.

The groups can be easily added by creating new fields as group dividers and dragged to the right place.

what do you think about this approach of grouping fields?

**Comment by pixelkicks** [comment]
Is this ever going to be completed and released?  This thread was originally created on 25 Feb 2012.

**Comment by sc0ttkclark** [comment]
@pixelkicks yes, we're in heavy development with the CMB2 team, already have some major work headed it's way into CMB2 to support nesting fields in multiple levels of group fields (their loop fields).

Appreciate your patience, we're on the home stretch!

**Comment by sc0ttkclark** [comment]
I can also say that it has been and has always been a money issue, the amount of work involved in making this happen while also providing the project with support and maintenance fixes has been a challenge. Loop fields in Pods required a lot of changes to the codebase, all of which is in Pods 3.0 now. And we're going to be using CMB2 to power 3.0 too, so we cut out a bunch of work we no longer have to do, while strengthening both projects.

I'm personally putting up my own money to fund @pglewis through the rest of the month and into mid-October. He's been on fire with the work he's been doing with the CMB2 refactor to support nested fields. We're also involving @wpscholar for the Backbone stuff, which is another thing I'm putting in personally. There's a lot I've done for Pods and now CMB2 to push them both forward, all of which we put out for free to the world. That's something we're all very proud of.

Anyone who wants Loop fields in Pods core, who hasn't donated to the project yet, could do a great service towards making this happen sooner through becoming a [Friend of Pods](https://pods.io/friends-of-pods/) through one-time or recurring donations. Every bit helps, I'm putting in the rest myself because it continues to be an important feature in Pods 3.0 that's holding up the release.

**Comment by pixelkicks** [comment]
Isn't Pods funded by Automattic? Aren't they stumping up enough moolah?

**Comment by sc0ttkclark** [comment]
Automattic is a Pillar sponsor through Friends of Pods. Pods isn't fully funded by Automattic, it only goes about 70% of the way towards the expenses of keeping @pglewis @jimtrue and @Shelob9 paid and with lights on, with a very modest income they generously work with for us because they all believe in our cause. We still rely on our other Friends of Pods to help make up the difference, but we're not a fully funded project yet. @pglewis normally ends up doing some side work in the name of Pods and having clients pay Pods directly, or I eat more of my credit cards up.

**Comment by StaggerLeee** [comment]
Make it as commercial plugin and charge for it then ?
Anything just to get away from Advanced Custom Fields plugin and "one man" developing.

I have all best to say about him, but made once fatal mistake with K2 in Joomla, and later had more, more luck than knowledge to convert all my websites to Joomla Articles manager, and later to the WordPress.

**Comment by sc0ttkclark** [comment]
Pods is a free plugin, we have no intentions of becoming a commercial plugin or having a pro version.

**Comment by pixelkicks** [comment]
I'd be happy with a freemium model. I would happily pay for features such as this.

Keep the core features in the free plugin, and include extras in a paid for version. What's wrong with that model?  Keep the costs low enough and you won't see many complaints. Users would have a better supported product and one with faster updates.

**Comment by StaggerLeee** [comment]
Not Pods. Make repeater field together with grouping as third party plugin.
What is a shame in it ? I dont understand.

I already have PRO licence for ACF, but honestly live in fear using it so extensively on all my websites.

**Comment by sc0ttkclark** [comment]
A loop field type has to be part of Pods core, it will not be a premium add-on. We're nearly there, just hold on and we'll have CMB2 fully integrated with Pods 3.0 and the world will be merry and everything is free.

**Comment by pixelkicks** [comment]
In 2015?

**Comment by sc0ttkclark** [comment]
Or bust

**Comment by sc0ttkclark** [comment]
@StaggerLeee and @pixelkicks -- If you're interested in paying for a commercial plugin solution, you should totally become a Friend of Pods and help make this a reality. If you're already using Pods for client projects and depend on it, and really want to see loop fields happen, that is literally the best thing you can do right now. If you're hesitant because you're worried Pods 3.0 won't be released in 2015, then I could promise that if we don't release it in 2015 then I will refund all of your donations (hopefully you become a recurring monthly friend). I believe in what we have going here, we've finally got the right people in place to make this happen, it's just a matter of having them able to work on it uninterrupted.

**Comment by lkraav** [comment]
Thanks for reminding me @sc0ttkclark. We briefly chatted about optimizing the Friends landing page back in the day. I'm going to hop on the silver bandwagon now, but I'd like to somehow get invoices to my company. Donations are a hassle from a taxation perspective. No need to reply here and take this issue further off-topic, but I thought maybe other business-owners have the same concern. I'm looking you up on podswp slack next.

**Comment by kenwiesner** [comment]
As the ONLY Silver Friend of Pods, I encourage all of you who know and love
Pods to join me in support of the amazing work that Scott and the entire
team are doing. Let's be serious here guys, Pods IS a "premium plugin" the
difference is that instead of charging everyone a flat fee, the Pods team
is asking "if you use it and you like it, help fund its development." It's
a model that makes this invaluable tool easily accessible to anyone and
expands the user base making it a more powerful tool for everyone.
https://pods.io/friends-of-pods/friends-and-partners/

**Comment by pixelkicks** [comment]
Ok, it's not a lot, but we've just signed up as a Green member for $5 a month. Every little counts huh?

**Comment by sc0ttkclark** [comment]
Every little bit counts a LOT. $5 adds up over time, and if we could get 520 people to become a friend out of the current 30k+ sites using Pods (https://wordpress.org/plugins/pods/) then we'd be fully funded.

**Comment by StaggerLeee** [comment]
I donated small amount.
But, still would like to see some goodies as premium plugin(s).

I am convinced it would be enough money from licences to finish this job from issue here 2 years ago.

**Comment by sc0ttkclark** [comment]
To be fully funded, we'd need:

463+ Green friends OR
93+ Bronze friends OR
28+ Silver friends OR 
8+ Gold friends OR

(or a mixture of the above)

**Comment by sc0ttkclark** [comment]
With our 21 current friends combined, we have the equivalent of 1 gold friend.

**Comment by SGillessen** [comment]
As @lkraav said, I'd be more willing to pay on a recurring basis, if I could be invoiced. Makes it much easier on my tax reports.

**Comment by sc0ttkclark** [comment]
@SGillessen I sent you an e-mail to get more details on your company to add to our invoicing system. If anyone else needs invoices, we can definitely get those setup.

**Comment by jimtrue** [comment]
Yep, email jim@pods.io and we'll get you properly setup for invoicing your donations.

**Comment by jimtrue** [comment]
Not Pods. Make repeater field together with grouping as third party plugin.
What is a shame in it ? I dont understand.

I already have PRO licence for ACF, but honestly live in fear using it so extensively on all my websites.

[--- Commented from Asana.com, linked by Unito.io
#commenter Jim True
---[aa]

**Comment by george-fb** [comment]
You said 3.0 will be out this year. Can you narrow it down some more?

Currently i'm using custom fields suite but i recently noticed that deleting an item from the loop in CFS doesn't  work as i expected, which makes it complicated for my needs.

If you could take a look and tell me if pods will handle it the same or not. 
Here is what i mean:
https://wordpress.org/support/topic/deleting-item-from-lop?replies=1

Keep up the awesome work, pods is one of the most powerful plugins for wordpress.

**Comment by szepeviktor** [comment]
GIMP developers say:
![ready](http://www.gimpusers.com/system/tutorial_steps/3072/itsdonewhenitsdone-thumb.jpg)

**Comment by nicdford** [comment]
@szepeviktor it doesn't take making $55k/yr to help, every little bit counts! At the very least, get yourself a sticker!

**Comment by sc0ttkclark** [comment]
Just following up, CMB2 is already in progress with the nested fields refactor to support what we need for Loop fields. The PHP side of this is complete and is awaiting review by @jsternberg. And @wpscholar is in progress with the Backbone refactor as the second part of that process right now. We're all pushing for 3.0 release this year, but with those parts up in the air, we can't commit to a specific date just yet. Will continue updating this ticket as we reach milestones on development.

**Comment by iCristiano** [comment]
Any update? 2016 so?

**Comment by quasel** [comment]
Due to some recent events and the Desicions to part ways with CMB2 http://pods.io/2015/11/12/pods-3-0-and-cmb2-parting-ways/ things change things a bit  http://pods.io/docs/comparisons/loop-fields-custom-vs-relationships/ - a more detailed post about the the roadmap will be relased in the next days

**Comment by sc0ttkclark** [comment]
Check our blog in just a few minutes :)

**Comment by waldbach** [comment]
sweet

**Comment by sc0ttkclark** [comment]
http://pods.io/2015/12/26/new-path-pods-3-0/

**Comment by JohnGraffio** [comment]
Loop fields / nested fields in Pods 3.0 will behave like the old Pick operating system multi-values, whose current incarnation is called OpenInsight. I used to program in that environment; when you get Pods to do that, it will be the most significant plugin available for WordPress. Can't wait.

**Comment by JoryHogeveen** [comment]
Hi all! I was just wondering... since I've used ACF for loop fields. When Pods has its own loop field. How will pods store this data? Will it be similar to ACF ( `**parentFieldName**_**row**_**childFieldName**` )?
I hope so since that will make it a lot easier to convert my websites to only use Pods :).

**Comment by StaggerLeee** [comment]
From 0% to 100% how much is done approximately ?

**Comment by sc0ttkclark** [comment]
Just posting an update on this, we've been working hard on our new Backbone-powered fields in #3364, specifically for #3311 and #3309.

I've been playing around with the alpha version of Pods 2.7 and it's really fantastic. Flexible relationships are going to be way better for structure than Loop fields could ever be, but this is the logical first step. So if you're looking for a % completion amount, I'd say we're about 50% right now, because 25% will be finishing up our cool new modal form syncing with the Backbone fields, and the remaining 25% will be applying this cool stuff to the Loop field type which exists as entirely custom data (not relational data that lives on their own records).

**Comment by StaggerLeee** [comment]
Give as info when 2.7 - 3.0 will be upgradeable via WordPress backend. RC, not Alpha, Beta. So we can start using it and report back problems.

**Comment by pglewis** [comment]
2.7 has been my top priority since the day I started on it.  If I could give you a hard and fast date I would have already done so.  The scope changes, unexpected road-blocks show up, real-life intervenes, etc.  Any estimate I give you would be pulling a date out of a hat that I cannot guarantee will be accurate, like many software estimates.  The only answer I have any time you ask is: as soon as we possibly can.

**Comment by StaggerLeee** [comment]
O man, just say it. I promise I wont pull you to the Court od Justice in EUSSR Brussels.  :)

**Comment by pglewis** [comment]
Let's please leave any and all political references out of GitHub, this is not the place.

**Comment by kochhase** [comment]
sorry I have waited over 1 Year of loop fields. Now I have to switched to toolset (types). There development is fast and the missing struktur for many to many relation will come end of the year.

**Comment by kelfy** [comment]
@kochhase You just waited a year but I've been waited many years. Look at the date of my first comment on this thread. 
The only advantage of Pods is allow you to create advanced custom fields which let you to store in a seperate table  instead of WP standard one. Other then that, ACF, Toolset, Custom Post Type UI..etc are doing great job.

**Comment by JoryHogeveen** [comment]
@kochhase @kelfy 
ACF would be your best pick for now if you need loop fields (though it is a premium feature). I would not recommend Toolset though.
Also note that Pods and ACF work very well together. So for example you can use Pods as your base for all your objects (CPT's etc.) and basic fields and use ACF for some extra features for example.

**Comment by kelfy** [comment]
@JoryHogeveen That's what I do too. ACF+Pods almost for all my projects over the last 2 years

**Comment by quasel** [comment]
@kelfy Sorry that it took longer then any of us expected maybe you did read some of [the reasons](http://pods.io/2015/11/12/pods-3-0-and-cmb2-parting-ways/ )

Pod's already has some sort of "LOOP" it's just stored in an related CPT and we are adding an UI to make it simpler / more usefull and call it "flexible relationships" #3309 - you can test an ["Alpha" ](https://github.com/pods-framework/pods/tree/feature/%233309 )

**Comment by szepeviktor** [comment]
#### As for mixing Toolset and ACF with Pods

Toolset's code quality is *low*, no better word for it.
ACF is a one man show.

Pods is an enterprise level plugin with [tests](https://github.com/pods-framework/pods/tree/2.x/tests/phpunit).

**Comment by kochhase** [comment]
Toolset's code quality is **was** low. that was before Toolset 2.
Now its good for non php user and has fast devolpment for the future. There are more than 8 developer working for it.

**Comment by StaggerLeee** [comment]
With this nonchalant attitude you are making it difficult for other developers. How it is possible ACF has more third party addons than Pods ? How on earth.

**Comment by szepeviktor** [comment]
@StaggerLeee 

> Pods is an enterprise level plugin with tests.

That means it is used where stability is critical.
ACF's client are - I think - non-technical people. They see only the screen.
I usually partly audit the plugins and contact the author also.

**Comment by StaggerLeee** [comment]
Maybe you did it difficult for yourself from beginning. Let us say this way. Why should you lose time on Repeater (Loop) field, Group addon, etc... Why not make it easy to third party developers to make it.
I know now it late, just chit-chat.

Something like ACF has for instance Table field as third party addon.
I dont know any third party addon extending Pods field types. Maybe there is but I missed it.

**Comment by quasel** [comment]
well this goes a bit offtopic - to answer you last Question - Pods is and was a more developer centric tool and has less of the shiny ui for the frontend. Regarding integrations it's all there for people to use, CIO, Gravity, iThemes, Easy Pods,... last week e.g we got contacted by AdminColumns Pro they are working to fully support pods including relationships.

And the beauty you can combine Plugins to get the best of both - after all Pod's is free and opensource and sponsored by [well known players](https://pods.io/friends-of-pods/friends-and-partners/) out of the WordPress Community!

**Comment by jimtrue** [comment]
This discussion is devolving into a bashing this plugin and a Pods vs ACF vs Types discussion. I'm not seeing anything posted towards moving the project or this phase of the project forward. Non-constructive criticism or comparisons don't really help anyone, they just become a discussion of 'why isn't this done yet?' and 'why haven't you done what this other plugin has done?' None of which are helpful or, again, move the project forward.

Sorry to come down with the 'ban' hammer, but if there is any more discussion that isn't _constructive_ towards providing code contributions, code suggestions or actual support in the form of documentation, help to other users with workarounds (with examples) or donations to [Friends of Pods](http://pods.io/friends-of-pods/) so we can actually afford to hire additional developers to move the project forward, we will ask GitHub to remove your ability to comment on this project because you have become what the internet defines as a 'troll'.

This is an open source, free plugin with free support. Our driving focus has always been to provide a framework that provides a fairly painless entry into the the Custom Post Types, Fields and Taxonomy world of WordPress to take your content beyond just Posts and Pages. We also have a very stable back-end that allows for extremely powerful enterprise level solutions. Loop fields have 'on the back-end' been in Pods since version 2.0 since we support relationship fields natively between Post Types, WordPress Objects and Taxonomies both one to many and many-to-many. What we're missing is the UI to provide that simplicity in the back-end. 

If you want to get involved in helping with that UI, provide code, download the feature branch (feature/2.7) and jump into the discussion in our [Slack Chat](http://pods.io/chat/) on the #core channel. If you want to keep criticizing the project without providing actual, workable solutions, please step off of this discussion.

**Comment by StaggerLeee** [comment]
I know all of this. Difficult to say something without offending someone.
If whole WordPress is managed as Pods instead of 47.000 addons it would have maybe 5000 of them.
Does it mean WordPress core code is of low quality ?

-----------------------------
For instance this is made yesterday. I dont know what you are doing, but maybe time to lift heads above code. I know it is high quality code, no need to repeat it.
![Image of Yaktocat](https://s22.postimg.org/65xg5dyc1/2016_10_03_140243.jpg)

**Comment by szepeviktor** [comment]
You may open a PR with:
```css
#side-sortables .form-table tr.pods-field:last-child > th {
    widtvh: 150px;
}
``

**Comment by sc0ttkclark** [comment]
We've got a *separate* process going on for responsive forms, so that's off topic.

Thanks for your patience so far, we're closer than ever before to a Loop Field Type with our work in Pods 2.7, which changes all of our form field types over to Backbone + Marionette JS. It's pretty cool stuff what @pglewis has managed to do with it so far.

I think it's best to temporarily lock this thread, that doesn't close it, but allows us to focus on completing the work needed for Pods 2.7 which is basically Loop Fields but with relational objects. I'll unlock the thread after 2.7 is released as this is taking time/resources away from what's most important (and echoing what Jim said): moving the project forward.

**Comment by sc0ttkclark** [comment]
Repeatable Fields (https://github.com/pods-framework/pods/issues/1095) is in Pods 2.9, you can check out Pods 2.9 Beta 2 at https://github.com/pods-framework/pods/releases/tag/2.9.0-beta-2
